### PR TITLE
OTG Test - TE-11.3: Backup NHG: Actions

### LINKED ISSUE #1005: OTG: TE-11.3: Backup NHG: Actions

### PR DESCRIPTION
**Pull request by octpetre**
OTG conversion of _TE-11.3: Backup NHG: Actions_
Fixes #1005

### CODE REVIEW

File: feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string, dstPfx string, dst *attrs.Attributes) gosnappi.Flow {
+
+	flow := gosnappi.NewFlow().SetName(name)
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{atePort1.Name + ".IPv4"}).SetRxNames([]string{dst.Name + ".IPv4"})
+	ethHeader := flow.Packet().Add().Ethernet()
+	ethHeader.Src().SetValue(atePort1.MAC)
+	outerIPHeader := flow.Packet().Add().Ipv4()
+	outerIPHeader.Src().SetValue(atePort1.IPv4)
+	outerIPHeader.Dst().Increment().SetStart(dstPfx).SetCount(1)
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go:318]: This is just equivalent to the `SetValue(dstPfx)` right? (I see that it's similar in the ATE test as well, not sure of the intent there.)

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+	if want := oc.Interface_OperStatus_DOWN; operStatusAfterShut != want {
+		t.Errorf("Get(DUT port2 oper status): got %v, want %v", operStatus, want)
+	}
+
+	t.Log("Validate traffic passes through port3")
+	validateTrafficFlows(t, args.ate, args.top, backupFlow, baselineFlow)
+}
+
+// createFlow returns a flow from atePort1 to the dstPfx.
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string, dstPfx string, dst *attrs.Attributes) gosnappi.Flow {
```
[the remark this answers is not in the document]
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go:309]: Removed

File: feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go (hunk 1 above)
Comment (jasdeep-hundal) [reply feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go:318]: Will resolve outside of this PR.

File: feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jasdeep-hundal) [reply feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go:309]: @octpetre : Not seeing the changes in the latest code, may need another push here.

[the remark this answers is not in the document]
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go]: Done

File: feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+	if want := oc.Interface_OperStatus_DOWN; operStatusAfterShut != want {
+		t.Errorf("Get(DUT port2 oper status): got %v, want %v", operStatus, want)
+	}
+
+	t.Log("Validate traffic passes through port3")
+	validateTrafficFlows(t, args.ate, args.top, backupFlow, baselineFlow)
+}
+
+// createFlow returns a flow from atePort1 to the dstPfx.
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, name string, dstPfx string, dst *attrs.Attributes) gosnappi.Flow {
```
Comment (jasdeep-hundal) [inline feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go:309]: Can still remove `t`

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go (hunk 3 above)
Comment (octpetre) [reply feature/experimental/gribi/otg_tests/backup_nhg_action/backup_nhg_action_test.go:309]: Removed

**Review by jasdeep-hundal — APPROVED** [review]

**Pull request merged**